tube/spark/es_writer.py

```diff
@@ -1,54 +1,178 @@
 import json
+import re
 
-from elasticsearch import Elasticsearch, client
+from elasticsearch import Elasticsearch, client, helpers
 
 from tube.utils import generate_mapping
 from tube.spark.plugins import post_process_plugins, add_auth_resource_path_mapping
-
 from tube.spark.spark_base import SparkBase
+from enum import IntEnum
 
 
 def json_export(x):
     x[1]['node_id'] = x[0]
     return (x[0], json.dumps(x[1]))
 
 
+def get_index_name(index, version):
+    return '{}_{}'.format(index, version)
+
+
+def get_backup_index_name(index, version):
+    return '{}_{}'.format(version, index)
+
+
+def get_backup_alias(index):
+    return '{}_backup'.format(index)
+
+
+def get_backup_version(index_name):
+    res = re.match('.?([0-9]+)*', index_name)
```

Comment:
Why the regex is so complex? Won't it result in the same as: `.?([0-9]*)`?

Reply:
need to have + because we need the number appear at least once

Reply:
Then `.?([0-9]+)`? I doesn't fully understand the need of both `+` for digits and `*` for the whole group. There can be multiple groups?

Reply:
* is used to point that the whole group appear at the beginning of the string and we don't care the remaining of the string